### DISCUSSION TITLE
New branch

### DISCUSSION THREAD

**Discussion by NovaLeg** [community/community#144936] [New to GitHub]
How do you create a new branch in GitHub?

**Comment by Amrit1604** [comment]
Method 1: Create a Branch Directly on GitHub (Web Interface)

Go to the Repository: Open your project on GitHub.
Click the Branch Dropdown: On the top left, click the dropdown where it says "main" (or the current branch name).
Type a New Branch Name: Enter the name for your new branch (e.g., feature-new-page).
Create the Branch: Press "Create branch", and your new branch will be created based on the current branch (usually main).

Method 2: Create a Branch Locally Using Git
If you’re using Git on your computer, follow these steps:

Navigate to Your Repository:


`cd your-repo-name
`Make Sure You’re on the Right Branch:


```
git checkout main
git pull origin main
```
Create a New Branch:


`git checkout -b new-branch-name
`This command does two things:
Creates a new branch called new-branch-name.
Switches you to that new branch.
Push the New Branch to GitHub:


`git push origin new-branch-name
`Check Your Branch:


`git branch
`This shows all the branches. The current branch will be highlighted.

**Reply by ghostinhershell** [reply]
Hi @NovaLeg 👋🏾,It looks like @Amrit1604's response was the answer to your question here. Can you please mark their response as the answer? When you mark a question as an answer, GitHub will highlight the comment and replies to the comment to help visitors quickly find the answer.
![image](https://github.com/user-attachments/assets/b89e7161-c2d1-47e2-a058-242df633e6b2)
